hack/get_ci_vm.sh

```diff
@@ -90,17 +82,31 @@ show_usage() {
 }
 
 get_env_vars() {
-    python -c '
-import yaml
+    # Deal with both YAML and embedded shell-like substitutions in values
+    # if substitution fails, fall back to printing nakid env. var as-is.
+    python3 -c '
+import yaml,re
 env=yaml.load(open(".cirrus.yml"), Loader=yaml.SafeLoader)["env"]
-keys=[k for k in env if "ENCRYPTED" not in str(env[k])]
+rx1=re.compile(r"\$(\w+)")
+rx2=re.compile(r"\$\{(\w+)\}")
+rep=r"{\1}"
+class ReIterKey(dict):
+    def __missing__(self, key):
+        return "${0}".format(key)
```

Comment:
Under what circumstances would this be necessary? Shouldn't it always be an error to reference an undefined variable?

Reply:
Sadly no.  Some variables are only available by Cirrus-CI at runtime.  This script doesn't pretend to perfectly duplicate their YAML env-var-like parser.  It's perfectly fine to reference one of these runtime-only variables in the YAML for Cirrus.  Nobody can can predict if/when might be added.  I don't want that to break this script :confused: 

(Why having a more robust and unit-tested `hack/get_ci_vm.sh` script would be nice)

Hmmm...how to comment that concisely...

Reply:
...okay I think I got it.